diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4530642..913ece5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -169,11 +169,6 @@ if(GUI_ENABLED)
     endif()
 endif()

-if(OPENGL_ENABLED AND NOT GUI_ENABLED)
-    message(STATUS "Disabling GUI will also disable OpenGL")
-    set(OPENGL_ENABLED OFF)
-endif()
-
 if(CGAL_FOUND)
     list(APPEND CGAL_LIBRARY \${CGAL_LIBRARIES})
     message(STATUS "Found CGAL")
diff --git a/src/feature/sift.cc b/src/feature/sift.cc
index 07d27e6..43882fe 100644
--- a/src/feature/sift.cc
+++ b/src/feature/sift.cc
@@ -36,7 +36,7 @@
 #include <memory>

 #include "flann/flann.hpp"
-#if !defined(GUI_ENABLED) && !defined(CUDA_ENABLED)
+#if defined(OPENGL_ENABLED)
 #include "GL/glew.h"
 #endif
 #include "SiftGPU/SiftGPU.h"
